### DISCUSSION TITLE
suuport different version of module-federation.manifest.json in assets

### DISCUSSION THREAD

**Discussion by ahmed-a-ibrahim** [nrwl/nx#23371] [Feature Requests]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

Ability to have different version of module-federation.manifest.json like module-federation.manifest.prod.json, module-federation.manifest.stage.json and auto reader by the webpack based on the mode

**Comment by Coly010** [comment]
This kind of defeats the purpose of Dynamic Module Federation.

Ideally, your module-federation.manifest.json file should be replaced during CD or fetched from an external location, rather than being part of your build. 

the idea behind it is that you build the application once, and can deploy the same artifact to any environment and it will continue to work.
